create initial workflow

Hamza-Mos opened the pull request:
## Description ✏️

Closes #xxx

Describe what this PR does.

- Bullet points are helpful.
- Screenshots are helpful, if applicable.

## Type of Change 🐞

- [ ] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [ ] I have done a self-review of my code.
- [ ] I have manually tested my code (if applicable).
- [ ] I have added/updated any relevant documentation (if applicable).

Hamza-Mos commented:
@tomas-salgado @ramiAbdou this wasn't explicitly described in the original issue but I added an additional case where an issue is not assigned for 2+ weeks (screenshot of what that looks like is in my PR description). This normally doesn't happen for Oyster though since the issues are constantly being assigned 😂

Let me know what you think

ramiAbdou commented on .github/workflows/issue-management.yml:
```diff
@@ -0,0 +1,87 @@
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
```
Nit: Is it possible to move all of this script logic to a separate JS file and import that JS file here? Thinking we could put this in like a `.github/scripts` folder. Feel like that could make it easier to update/maintain since the formatting in a `.yml` file is the equivalent of a long string haha.

ramiAbdou commented on .github/workflows/issue-management.yml:
```diff
@@ -0,0 +1,87 @@
+            const threeWeeksAgo = new Date(Date.now() - 21 * 24 * 60 * 60 * 1000);
+            const botLogin = 'github-actions[bot]';
+
+            const issues = await github.rest.issues.listForRepo({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              state: 'open'
+            });
```
I was looking at the GitHub API [documentation](https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#list-repository-issues) and it seems like this will also return pull requests? It's very confusing that they'd consider all issues as PRs but worth verifying that. We may have to do some additional filtering to ensure that we're only looking at issues and not PRs.

ramiAbdou commented on .github/workflows/issue-management.yml:
```diff
@@ -0,0 +1,87 @@
+                    body: 'This issue has been inactive for 3 weeks and has been unassigned. Feel free to pick it up again when you\'re ready to work on it.'
+                  });
+                } else if (!warningAlreadyGiven) {
+                  await github.rest.issues.createComment({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    issue_number: issue.number,
+                    body: 'This unassigned issue has been inactive for 3 weeks. Please consider assigning it to someone or updating its status.'
+                  });
+                }
```
I appreciate you looking ahead, but I think we can remove this case. We sometimes create top-level "project" issues with a bunch of subtasks and we don't necessarily want to assign the top-level issue to anyone, so would be a tad annoying to get pinged about this.

Also, like you mentioned also, issues go very fast around here! 😂

ramiAbdou commented on .github/workflows/issue-management.yml:
```diff
@@ -0,0 +1,87 @@
+                // warning if inactive for 2 weeks
+                const warningMessage = isAssigned
+                  ? 'This issue has been inactive for 2 weeks. Please provide an update or it may be unassigned in 1 week.'
+                  : 'This unassigned issue has been inactive for 2 weeks. Please consider assigning it to someone or updating its status.';
```
Same comment here, think we can remove this!

ramiAbdou commented on .github/workflows/issue-management.yml:
```diff
@@ -0,0 +1,87 @@
+              state: 'open'
+            });
+
+            for (const issue of issues.data) {
```
Is it possible to filter out issues that have an open PR linked to it? I think those issues should be considered active regardless of the comments on the issue itself!

ramiAbdou reviewed (changes requested):
This looks really solid @Hamza-Mos! Truth be told, I'm scared of implementing GitHub Action workflows, I always have a hard time with them but you knocked this out super quickly. 💯 Appreciate you!

Just a couple comments/questions, almost ready to get in!

Hamza-Mos commented:
> This looks really solid @Hamza-Mos! Truth be told, I'm scared of implementing GitHub Action workflows, I always have a hard time with them but you knocked this out super quickly. 💯 Appreciate you!
> 
> Just a couple comments/questions, almost ready to get in!

Appreciate the feedback - have addressed the comments and should be good for another review now @ramiAbdou 🫡

ramiAbdou reviewed (approved):
This looks great, @Hamza-Mos! Thank you for tackling this! 🚀
